### PR TITLE
Fixes #2230

### LINKED ISSUE #2230: Cyrillic domains https problem
##### Steps to reproduce the problem:

1. Start http proxy
2. Go to https://фонд-достоинство.рф
3. See the problem Certificate Verification Error for фонд-достоинство.рф: hostname 'фонд-достоинство.рф' doesn't match 'xn----dtbebu0aecead5adket.xn--p1ai'

##### System information

Mitmproxy version: 2.0.1 (release version) 
Python version: 3.6.1
Platform: Darwin-15.6.0-x86_64-i386-64bit
SSL version: OpenSSL 1.1.0e  16 Feb 2017
Mac version: 10.11.6 ('', '', '') x86_64

### PR DESCRIPTION
**Pull request by ujjwal96**
SNI hostnames should be in ASCII only form.
This is done by using A-labels defined in RFC 5890. 
Conversion to ACE(ASCII-Compatible Encoding) is described in RFC 3490

closes #2245

### CODE REVIEW

**Comment by Kriechi** [comment]
Please add a test that actually checks the correct encoding.

File: test/mitmproxy/net/test_tcp.py [hunk 1]
```diff
@@ -387,6 +387,13 @@ def test_echo(self):
             assert c.sni == "foo.com"
             assert c.rfile.readline() == b"foo.com"
 
+    def test_idn(self):
+        c = tcp.TCPClient(("127.0.0.1", self.port))
+        with c.connect():
+            c.convert_to_ssl(sni="кц.рф")
+            assert c.sni.encode("idna").decode("ascii") == "xn--j1ay.xn--p1ai"
```
Comment (mhils) [inline test/mitmproxy/net/test_tcp.py]: This test would still pass without the changes above, right?

Comment (ujjwal96) [inline test/mitmproxy/net/test_tcp.py]: Yes it would. This is what I changed in `convert_to_ssl`  which is why I thought of writing it
Should I remove it?

Comment (mhils) [inline test/mitmproxy/net/test_tcp.py]: If I revert your changes in tcp.py, this still passes for me.

**Pull request merged**